docs/dev_guide/maintainer_workflow.rst

```diff
@@ -0,0 +1,124 @@
+.. _maintainer-workflow:
+
+Workflow for Maintainers
+========================
+
+This page is for maintainers those of us who merge our own or other
+peoples' changes into the upstream repository.
+
+Being as how you're a maintainer, you are completely on top of the basic stuff
+in :ref:`version_control`.
+
+Integrating changes
+-------------------
+
+Whenever possible, merge pull requests automatically via the pull request manager on GitHub. Merging should only be done manually if there is a really good reason to do this!
+
+Please try to make sure that pull requests do not contain a messy history with merges, etc. If this is the case, then you have two options. To use the rebase merge on Github or to follow the manual instructions, and make sure the fork is rebased to tidy the history before committing. The second option should only be used on complex pull requests.
+
+
+Using Milestones and Labels
+===========================
+
+These guidelines are adapted from `similar guidelines <http://docs.astropy.org/en/stable/development/workflow/maintainer_workflow.html#using-milestones-and-labels>`_
+.. _milestones-and-labels:
+followed by AstroPy:
+
+* All open pull requests should have a milestone.
+
+* Only confirmed issues that are release critical or for some other reason should be addressed for a release, should have a milestone.
+
+* In general there should be the following open milestones:
+
+  * The next bug fix releases for any still-supported version lines; for example if 0.4 is in development and
+    0.2.x and 0.3.x are still supported there should be milestones for the next 0.2.x and 0.3.x releases.
+
+  * The next X.Y release, i.e. the next minor release; this is generally the next release that all development in
+    master is aimed toward.
+
+  * The next X.Y release +1; for example if 0.3 is the next release, there should also be a milestone for 0.4 for
+    issues that are important, but that we know won't be resolved in the next release.
+
+* When in doubt about which milestone to use for an issue, use the next minor release, it can always be moved once
+  it's been more closely reviewed prior to release.
+
+* Issues that require fixing in the mainline, but that also are confirmed to apply to supported stable version lines
+  should be marked with a ``Affects Release`` label and the corresponding supported stable version label ``v0.4.x``.
+
+Using Projects
+==============
+
+Currently our use of projects is not fully formalized.
+We create projects for the next major releases and organize what pull requests or issues we deem to be blockers for a release or items we want to have in the next release but are not blockers.
+How this interacts with milestones, is still undecided.
```

Comment:
I would say milestones are for releases, and projects are for projects, e.g., summer of code projects, new coordinates transformations, and yes, for sure, releases!